cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/parser/YamlParser.java

```diff
@@ -39,7 +41,21 @@ public ParsedSlang parse(SlangSource source) {
             parsedSlang.setName(source.getName());
             return parsedSlang;
         } catch (Throwable e) {
-            throw new RuntimeException("There was a problem parsing the YAML source: " + source.getName() + ".\n" + e.getMessage(), e);
+            String errorMessage = e.getMessage();
+            if (e instanceof ScannerException && (errorMessage.startsWith("mapping values") || errorMessage.startsWith("while scanning a simple key"))){
+                errorMessage += "Probably did not provide (key: value) pair or missing space after colon(:)";
+            }
+            else if (e instanceof ConstructorException && errorMessage.startsWith("Cannot create property")){
+                if (errorMessage.contains("Unable to find property")){
+                    String truncatedErrorMessage = errorMessage.substring(errorMessage.indexOf("Unable"), errorMessage.indexOf("on class"));
+                    String undefinedProperty = truncatedErrorMessage.substring(truncatedErrorMessage.indexOf("\'")+1, truncatedErrorMessage.lastIndexOf("\'"));
+                    errorMessage += "Property \'" + undefinedProperty + "\' is not supported by CloudSlang. Check that \'" + undefinedProperty + "\' is indented properly.";
+                }
+                else if (errorMessage.contains("No single argument constructor found for interface java.util.Map")){
+                    errorMessage += "Probably did not provide (key: value) pair or missing space after colon(:)";
+                }
+            }
+            throw new RuntimeException("There was a problem parsing the YAML source: " + source.getName() + ".\n" + errorMessage, e);
```

Comment:
@tudorlesan In the message thrown in the runtime exception, the exception error message appears twice - once from the e.getMessage() and once from the errorMessage.
You should only throw the exception message once.